packages/edit-site/src/components/block-editor/index.js

```diff
@@ -44,38 +47,45 @@ export default function BlockEditor( { settings: _settings } ) {
 			},
 		};
 	}, [ canUserCreateMedia, _settings ] );
-	const [ content, _setContent ] = useEntityProp(
+	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
 		'postType',
-		'wp_template',
-		'content'
+		settings.templateType
 	);
-	const initialBlocks = useMemo( () => {
-		if ( typeof content !== 'function' ) {
-			const parsedContent = parse( content );
-			return parsedContent.length ? parsedContent : undefined;
-		}
-	}, [] );
-	const [ blocks = initialBlocks, setBlocks ] = useEntityProp(
-		'postType',
-		'wp_template',
-		'blocks'
+	const setActiveTemplateId = useCallback(
+		( newTemplateId ) =>
+			setSettings( ( prevSettings ) => ( {
+				...prevSettings,
+				templateId: newTemplateId,
+				templateType: 'wp_template',
+			} ) ),
+		[]
 	);
-	const setContent = useCallback( ( nextBlocks ) => {
-		setBlocks( nextBlocks );
-		_setContent( serialize( nextBlocks ) );
-	}, [] );
 	return (
 		<BlockEditorProvider
 			settings={ settings }
 			value={ blocks }
-			onInput={ setBlocks }
-			onChange={ setContent }
+			onInput={ onInput }
+			onChange={ onChange }
+			useSubRegistry={ false }
 		>
 			<BlockEditorKeyboardShortcuts />
+			<URLPopover.LinkViewer.Fill>
+				{ useCallback(
+					( fillProps ) => (
+						<NavigateToLink
+							{ ...fillProps }
+							templateIds={ settings.templateIds }
+							activeId={ settings.templateId }
+							onActiveIdChange={ setActiveTemplateId }
+						/>
+					),
+					[ settings.templateIds, settings.templateId, setActiveTemplateId ]
+				) }
+			</URLPopover.LinkViewer.Fill>
```

Comment:
I'm having some trouble understanding what this does?

Reply:
It lets us add options to the link viewer, here that is the button to open the template it links to, if any.

Reply:
Still don't get it? How can we test to understand it more?

Reply:
Click on an internal link and see the button to edit the template it links to.

Reply:
I'm having trouble getting an example wiith this. That said, I'm not really confident about the new API (fill) added to achieve this. I also feel this should have deserved a dedicated PR in order to test it, discuss it properly and come up with the best solution.

Is this a feature that was asked for in FSE designs?

Reply:
Ahh, you're having trouble because links are now refactored to use the new `LinkControl`, so this `URLPopover` is not being used anymore. Shouldn't we deprecate it?

>Is this a feature that was asked for in FSE designs?

Yes